core/tests/gateway-migration-test/tests/migration.test.ts

```diff
@@ -0,0 +1,281 @@
+            await mintToAddress(baseToken, tester.ethWallet, tester.syncWallet.address, depositAmount * 10n);
+        }
+
+        const firstDepositHandle = await tester.syncWallet.deposit({
+            token: baseToken,
+            amount: depositAmount,
+            to: alice.address
+        });
+        await firstDepositHandle.wait();
+        while ((await tester.web3Provider.getL1BatchNumber()) <= initialL1BatchNumber) {
+            await utils.sleep(1);
+        }
+
+        const secondDepositHandle = await tester.syncWallet.deposit({
+            token: baseToken,
+            amount: depositAmount,
+            to: alice.address
+        });
+        await secondDepositHandle.wait();
+        while ((await tester.web3Provider.getL1BatchNumber()) <= initialL1BatchNumber + 1) {
+            await utils.sleep(1);
+        }
+
+        const balance = await alice.getBalance();
+        expect(balance === depositAmount * 2n, 'Incorrect balance after deposits').to.be.true;
+
+        // Wait for at least one new committed block
+        let newBlocksCommitted = await l1MainContract.getTotalBatchesCommitted();
+        let tryCount = 0;
+        while (blocksCommitted === newBlocksCommitted && tryCount < 30) {
+            newBlocksCommitted = await l1MainContract.getTotalBatchesCommitted();
+            tryCount += 1;
+            await utils.sleep(1);
+        }
+    });
+
+    step('Migrate to chain gateway', async () => {
+        if (direction == 'TO') {
+            await utils.spawn(`zkstack chain notify-about-to-gateway-update --chain ${fileConfig.chain}`);
+        } else {
+            await utils.spawn(`zkstack chain notify-about-from-gateway-update --chain ${fileConfig.chain}`);
+        }
+        // Trying to send a transaction from the same address again
+        await checkedRandomTransfer(alice, 1n);
+
+        if (direction == 'TO') {
+            await utils.spawn(
+                `zkstack chain migrate-to-gateway --chain ${fileConfig.chain} --gateway-chain-name ${gatewayChain}`
+            );
+        } else {
+            await utils.spawn(
+                `zkstack chain migrate-from-gateway --chain ${fileConfig.chain} --gateway-chain-name ${gatewayChain}`
+            );
+        }
+        await mainNodeSpawner.mainNode?.waitForShutdown();
+        // Node is already killed, so we simply start the new server
+        await mainNodeSpawner.killAndSpawnMainNode();
+    });
+
+    step('Check the settlement layer on l1 and gateway', async () => {
+        if (direction == 'TO') {
+            let gatewayInfo = getGatewayInfo(pathToHome, fileConfig.chain!);
+            let slAddressl1 = await l1MainContract.getSettlementLayer();
+            let slMainContract = new ethers.Contract(
+                gatewayInfo?.l2DiamondProxyAddress!,
+                ZK_CHAIN_INTERFACE,
+                gatewayInfo?.gatewayProvider
+            );
+            let slAddressSl = await slMainContract.getSettlementLayer();
```

Comment:
this a bit more correct

```suggestion
            let slAddressGW = await slMainContract.getSettlementLayer();
```